lib/cashu/proof.ts

```diff
@@ -0,0 +1,20 @@
+import type { Proof } from '@cashu/cashu-ts';
+import { getP2PKPubkeyFromSecret } from './secret';
+
+/**
+ * Get the pubkey that a list of proofs are locked to.
+ * @param proofs - The list of proofs to get the pubkey from.
+ * @returns The pubkey in the P2PK secrets
+ * @throws Error if there are multiple pubkeys in the list or if the secret is not a P2PK secret.
+ */
+export const getP2PKPubkeyFromProofs = (proofs: Proof[]) => {
+  const pubkeys = [
+    ...new Set(
+      proofs.map((p) => getP2PKPubkeyFromSecret(p.secret)).filter(Boolean),
+    ),
+  ];
+  if (pubkeys.length > 1) {
+    throw new Error('Received a token with multiple pubkeys');
```

Comment:
can this happen in practice? if yes should we just return all pubkeys found?

Reply:
Yea it can. A token could have proofs each locked to different pub keys. The reason that I throw an error is that if a token has multiple P2PK locks,  then  handling those proofs becomes more complicated.  I don't know exactly how that will work out yet.

In the case that there is only a single pub key in the proof set, all that we can do is check whether we have a key that can sign for the pubkey and if we have the private key, then we create signatures for all those proofs when we want to spend. 

When the proof set has multiple pubkeys, then some of those proofs might be spendable and some might not be by the private key we control. Even if we have a key for each pubkey we still would have to use the correct key to create the signatures.

I also just noticed this current implementation does not handle the case where some proofs have a P2PK secret and other just have a plain secret. The proofs with plain secrets would still be spendable, but should probably be separated out from the set of proofs with the P2PK lock.

Do you think that we should try to handle all the cases where the secrets are not consistent, or should we just enforce that if this function (`getP2PKPubkeyFromProofs`) is called, then the proofs have to all have P2PK secrets locked to the same pubkey. If there are multiple pubkeys or any plain secrets we throw an error.

Reply:
lets keep as is until we see how this works in action. it's hard for me to completely understand it atm without seeing how it is used exactly. we can always change this later if needed